googlemock/docs/translations/ko_KR/for_dummies.md

```diff
@@ -0,0 +1,452 @@
+
+**Tip:** 테스트코드 구현을 처음 시작하는 시점에는 expectation을 최대한 느슨하게 설정하는 것이 좋다. 쉽게 말해서 cardinality는 `Times(AnyNumber())`으로 하고 matcher는 `_`로 해놓고 시작하는 것이다. 그런 후에 개발을 진행하면서 점점 특화된 경우를 늘려가는 것이 유연성 측면에서 좋은 방법이다. 물론, "uninteresting call" 에 대해서는 굳이 이렇게 할 필요가 없겠지만, 관심대상이 되는 mock method에 대해서는 상당히 유용할 것이다. "uninteresting call"에 대해 궁금하다면 [Understanding Uninteresting vs Unexpected Calls](cook_book.md#uninteresting-vs-unexpected-를-구분하자)를 참조하자.
+
+#### Ordered vs Unordered Calls
+
+위에서 얘기한 것처럼 **특정(1개)** mock method에 여러개의 expectation이 지정되어 있는 상태에서 해당 mock method가 호출되면 reverse order로 expectation을 탐색하게 된다. 그런데 여기서 주의해야 할 점은 reverse order라는 것은 1개의 mock method에 지정된 expectation들을 탐색하는 순서일 뿐이며 우리는 아직 그들이 호출되어야 하는 순서를 지정한 적은 없다. 탐색순서와 호출순서가 서로 다른 것임에 주의하면서 이제부터는 mock method의 호출순서를 지정하는 방법에 대해 알아보자. 이 때는 1개의 mock method에 지정된 expectation들간의 호출순서뿐만 아니라 여러 mock method간의 호출순서를 지정하는 것도 포함한다.
+
+gMock에서는 expectation들의 호출순서를 지정하는 방법도 아래처럼 간단하게 구현할 수 있다.
+
+```c++
+using ::testing::InSequence;
+...
+TEST(FooTest, DrawsLineSegment) {
+  ...
+  {
+    InSequence seq;
+
+    EXPECT_CALL(turtle, PenDown());
+    EXPECT_CALL(turtle, Forward(100));
+    EXPECT_CALL(turtle, PenUp());
+  }
+  Foo();
+}
+```
+
+위 예제처럼 어떤 scope(block)를 만들고 scope 내부에 `InSequence` object를 생성하게 되면 해당 scope 내에 구현된 모든 expectation들이 *순서대로 호출되기를 기대한다*는 의미이다. 또한, 이 때의 호출순서는 reverse order가 아니며 위에서 아래 방향이 된다. 사용자가 몰라도 되는 부분이긴 하지만 gMock 내부적으로는 `InSequence` object의 constructor와 destructor를 통해서 이러한 호출순서를 검증하게 된다. (`InSequence` object의 이름이 무엇인지는 중요하지 않습니다.)
+
+결과적으로 위 코드는 `Foo()`라는 function이 내부에서 `turtle` object의 mock method 3개(`PenDown()`, `Forward(100)`, `PenUp()`)를 순서대로 호출하기를 기대한다고 해석할 수 있다. 따라서 3개 function이 모두 호출되었다고 하더라도 `PenDown()` -> `Forward(100)` -> `PenUp()`이라는 순서를 지키지 않았다면 테스트는 실패한다.
+
+만약, 관심대상인 function들을 단 하나의 흐름으로 묶는것이 아니라 여러 흐름으로 나눠서 호출순서를 지정하고 싶다면 어떻게 해야할까? gMock은 그러한 방법도 제공하고 있으며 관련된 내용은 [여기](cook_book.md#함수의-호출순서를-부분부분-지정하기)에서 확인하자.
+
+#### 모든 Expectation은 연결되어 있습니다. (Sticky Expectations)
+
+간단한 퀴즈를 풀어봅시다. `Turtle`이 원점에 도착한 횟수가 2번인지 확인하는 테스트를 구현하려면 어떻게 해야 할까? 즉, 테스트의 목적은 원점이동이 2회인지를 검증하는 것 외에 다른 것에는 관심이 없다고 가정해보자.
+
+먼저 스스로 풀어보고 아래 코드와 비교해보자.
+
+```cpp
+using ::testing::_;
+using ::testing::AnyNumber;
+...
+EXPECT_CALL(turtle, GoTo(_, _))  // #1
+     .Times(AnyNumber());
+EXPECT_CALL(turtle, GoTo(0, 0))  // #2
+     .Times(2);
+```
+
+위 코드와 같이 expectation을 설정한 다음에 원점으로 이동하라는 명령인 `GoTo(0, 0)`를 3번 호출하면 어떻게 될까? 일단, gMock은 reverse order로 expectation을 탐색하기 때문에 #2번 expectation을 먼저 확인한다. 그런데 #2번에서는 `GoTo()`가 정확히 2번 호출될 것이라고 기대하고 있기 때문에 3번째로 호출되면 테스트는 실패하게 될 것이다. (여기서 실패하는 이유는 [Using Multiple Expectation](#여러개의-expectations-사용하기)을 참고)
+
+이 예제는 사실 gMock의 **expectation들이 "sticky" 모드로 설정되어 있음**을 보여준다. "sticky" 모드는 호출횟수가 초과된 expectation들이 active 상태로 남아있는 모드를 의미한다. 이것은 매우 중요한 규칙인데 왜냐하면 다른 mocking framework들과 **구별되는** gMock의 특징이기 때문이다. 그럼 gMock에서는 왜 그렇게 했을까? 그 이유는 이러한 방식이 테스트를 구현하고 이해하는데 있어서 더 유리하다고 판단했기 때문이다.
+
+이유가 너무 간단한지도 모르겠다. 예제를 통해서 sticky 모드가 무엇인지 그리고 왜 유리한지에 대해 자세히 알아보자.
```

Comment:
nit: I guess "Simple?" refers to the previous quiz.

Reply:
Please check result here https://github.com/google/googletest/pull/2513/commits/2e9001784e73743363e34e347e89dbbe8def2322  
and let me know if there are some mistakes